### PR TITLE
Flag bulk print errors in dwp secondary state field

### PR DESCRIPTION
**Pull request by iguannago**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareServiceIt.java [hunk 1]
```diff
@@ -184,28 +197,26 @@ public void appealWithMrnDateOlderThan30Days_shouldGenerateDL16TemplateAndAndAdd
...
+        verify(evidenceManagementService).upload(any(), eq("sscs"));
         verify(ccdService).updateCase(any(), any(), any(), any(), eq("Uploaded dl16-12345656789.pdf into SSCS"), any());
         verify(bulkPrintService).sendToBulkPrint(any(), any());
 
         verify(ccdService).updateCase(any(), any(), eq(EventType.SENT_TO_DWP.getCcdType()), any(), eq(documentList), any());
     }
 
-    @Test(expected = NoMrnDetailsException.class)
+    @Test
     public void appealWithNoMrnDate_shouldNotGenerateTemplateOrAddToCcd() throws IOException {
```
Comment (jackmaloney) [inline src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareServiceIt.java]: Change description

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareServiceIt.java [hunk 2]
```diff
@@ -184,28 +197,26 @@ public void appealWithMrnDateOlderThan30Days_shouldGenerateDL16TemplateAndAndAdd
...
-            verifyNoMoreInteractions(evidenceManagementService);
-            verifyNoMoreInteractions(ccdService);
-            throw e;
-        }
+        evidenceShareService.processMessage(json);
+        verifyNoMoreInteractions(restTemplate);
+        verifyNoMoreInteractions(evidenceManagementService);
+        then(ccdService)
+            .should(times(1))
+            .updateCase(any(), any(), eq("sendToDwp"), any(), any(), any());
```
Comment (jackmaloney) [inline src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareServiceIt.java]: Can we capture the new `BulkPrintInfo` with expected output here?

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareServiceIt.java (hunk 1 above)
Comment (iguannago) [reply src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareServiceIt.java]: @jackmaloney , I think the name of the method is still valid. This method is still checking a case with no MrnDate and ensuring it is not generating the template and adding to CCD.

Comment (iguannago) [reply src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareServiceIt.java]: I renamed the method to capture we also update the case to flag the error.

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareServiceIt.java (hunk 2 above)
Comment (iguannago) [reply src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareServiceIt.java]: bulkPrintInfo is not an argument of that ccdService mock so we cannot capture. I have captured the caseData to ensure we are on the expected scenario and that therefore the case does not have mrn date and that we flag the error in the case and that the processMessage result is empty.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareService.java [hunk 3]
```diff
@@ -78,54 +84,99 @@ public EvidenceShareService(
...
+
+                log.info("Sending to bulk print for case id {}", sscsCaseDataCallback.getCaseDetails().getId());
+                caseData.setDateSentToDwp(LocalDate.now().toString());
+                return BulkPrintInfo.builder()
+                    .uuid(bulkPrintService.sendToBulkPrint(existingCasePdfs, caseData).orElse(null))
+                    .allowedTypeForBulkPrint(true)
+                    .desc(buildEventDescription(existingCasePdfs))
+                    .build();
+            }
+            return null;
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareService.java]: When this is null, should we not be flagging the error as a secondary state on the case? At the moment you are only flagging it if there is an exception thrown. I guess there could be a problem getting the template so probably need to capture this scenario as well.

Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareService.java]: ok. I was respecting the logic as it was previously, but yeah, it makes sense to flag the error for this case too.

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareServiceIt.java (hunk 2 above)
Comment (jackmaloney) [reply src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareServiceIt.java]: 👍

File: src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareService.java (hunk 3 above)
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareService.java]: done

**Pull request merged**